source/beamline/SGM/energy/SGMGratingTranslationStepControl.cpp

```diff
@@ -0,0 +1,89 @@
+#include "SGMGratingTranslationStepControl.h"
+
+#include "beamline/AMPVControl.h"
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+SGMGratingTranslationStepControl::SGMGratingTranslationStepControl(QObject *parent) :
+	AMPseudoMotorControl("Grating Translation Steps", "Steps", parent, "SGM Monochromator Grating Translation Steps")
+{
+	gratingTranslationStepPV_ = new AMPVwStatusControl("Grating Translation",
+													   "SMTR16114I1016:step:fbk",
+													   "SMTR16114I1016:step",
+													   "SMTR16114I1016:state",
+													   "SMTR16114I1016:emergStop",
+													   this,
+													   0.1,
+													   2.0,
+													   new AMControlStatusCheckerStopped(0));
+
+	addChildControl(gratingTranslationStepPV_);
+
+	setMinimumValue(8500);
+	setMaximumValue(62000);
+	setTolerance(10);
+	setDisplayPrecision(12);
+
+	updateStates();
+}
+
+bool SGMGratingTranslationStepControl::shouldMeasure() const
+{
+	return true;
+}
+
+bool SGMGratingTranslationStepControl::shouldMove() const
+{
+	return true;
+}
+
+bool SGMGratingTranslationStepControl::shouldStop() const
+{
+	return true;
+}
+
+bool SGMGratingTranslationStepControl::canMeasure() const
```

Comment:
Same note here with canMeasure()/canMove()/canStop() as for the SGMEnergyControl. If the gratingTranslationStepPV_ can only measure/move/stop sometimes, maybe include a check here: gratingTranslationStepPV_->canMeasure() (etc).